File: backend/app/schemas/document.py
from pydantic import BaseModel
from typing import Optional

class DocumentPage(BaseModel):
    page_number: int
    text: str


class DocumentMetadata(BaseModel):
    title: Optional[str] = None
    author: Optional[str] = None
    subject: Optional[str] = None


class Document(BaseModel):
    filename: str
    page_count: int
    metadata: DocumentMetadata
    pages: list[DocumentPage]

File: frontend/src/types/document.ts
export interface DocumentMetadata {
  title: string;
  author: string;
  subject: string;
}

export interface DocumentPage {
  page_number: number;
  text: string;
}

export interface Document {
  filename: string;
  page_count: number;
  metadata: DocumentMetadata;
  pages: DocumentPage[];
}